packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -574,8 +574,11 @@ import 'package:{{name}}/{{file}}';
 {{/methodChannelPlugins}}
 import 'package:flutter_web_plugins/flutter_web_plugins.dart';
 
-void registerPlugins([final Registrar? pluginRegistrar]) {
-  final Registrar registrar = pluginRegistrar ?? webPluginRegistrar;
+void registerPlugins() {
+  registerPluginsWithRegistrar(webPluginRegistrar);
+}
+
+void registerPluginsWithRegistrar(final Registrar registrar) {
```

Comment:
Do we know if we're building in null safe mode or not when we process this template? If we do, can we conditionally include/exclude the `?` depending on the mode?

Reply:
I couldn't find a simple and robust way to do so. Whether or not null safety is used is dependent on the environment.sdk range specified in the pubspec file, which is how I actually exercise this path for the unit test. This seems fine for a unit test, but it feels flimsy for the actual codegen that end users will run in production. Is it important that we do it as a single function call with an optional parameter rather than what I've done here?